extensions/wikia/NjordPrototype/scripts/Mom.js

```diff
@@ -0,0 +1,303 @@
+			$momBarContent.refresh();
+			$moms.refresh();
+			$moduleEditButton.refresh();
+			$moduleDiscardButton.refresh();
+			$moduleSaveButton.refresh();
+		};
+
+	if (window.wgUserName) {
+		initMom();
+	}
+	$mwContent.on('click', '.mom-edit-btn', function () {
+		var $moduleContainer = $(this).parents('.mom-module');
+		$moduleContainer.startThrobbing();
+
+		$.nirvana.sendRequest({
+			controller: 'NjordController',
+			method: 'getWikiMarkup',
+			type: 'GET',
+			data: {
+				articleTitle: $moduleContainer.attr('data-title')
+			},
+			callback: function (resp) {
+				$moduleContainer.removeClass('mom-hidden');
+				$moduleContainer.addClass('mom-edit');
+				$('.mom-save-btn', $moduleContainer).show();
+				$('.mom-discard-btn', $moduleContainer).show();
+
+				$('.mom-edit-btn', $moduleContainer).hide();
+				$('.mom-delete-btn', $moduleContainer).hide();
+				$('.mom-content', $moduleContainer).hide();
+				$('.mom-overlay', $moduleContainer).hide();
+
+				$('.mom-wiki-markup .wiki-markup', $moduleContainer).val(resp['wikiMarkup']);
+				$('.mom-wiki-markup', $moduleContainer).show();
+				$moduleContainer.stopThrobbing();
+			},
+			onErrorCallback: function () {
+				// TODO: handle failure
+			}
+		});
+	});
+
+	$mwContent.on('click', '.mom-discard-btn', function () {
+		var $moduleContainer = $(this).parents('.mom-module');
+		$moduleContainer.addClass('mom-hidden');
+		$moduleContainer.removeClass('mom-edit');
+
+		$('.mom-save-btn', $moduleContainer).hide();
+		$('.mom-discard-btn', $moduleContainer).hide();
+
+		$('.mom-edit-btn', $moduleContainer).show();
+		$('.mom-delete-btn', $moduleContainer).show();
+		$('.mom-overlay', $moduleContainer).show();
+		$('.mom-content', $moduleContainer).show();
+
+		$('.mom-wiki-markup', $moduleContainer).hide();
+	});
+
+	$mwContent.on('click', '.mom-save-btn', function () {
+		var $moduleContainer = $(this).parents('.mom-module');
+		$moduleContainer.startThrobbing();
+
+		$.nirvana.sendRequest({
+			controller: 'NjordController',
+			method: 'MainPageModuleSave',
+			type: 'POST',
+			data: {
+				pageTitle: $moduleContainer.attr('data-title'),
+				wikiMarkup: $('.wiki-markup', $moduleContainer).val()
+			},
+			callback: function (resp) {
+				$moduleContainer.addClass('mom-hidden');
+				$moduleContainer.removeClass('mom-edit');
+
+				$('.mom-save-btn', $moduleContainer).hide();
+				$('.mom-discard-btn', $moduleContainer).hide();
+
+				$('.mom-edit-btn', $moduleContainer).show();
+				$('.mom-delete-btn', $moduleContainer).show();
+				$('.mom-overlay', $moduleContainer).show();
+				$('.mom-content', $moduleContainer).html(resp['html']);
+				$('.mom-bar-info', $moduleContainer).html('content of this module might be outdated, refresh page for updated version');
+				$('.mom-content', $moduleContainer).show();
+
+				$('.mom-wiki-markup', $moduleContainer).hide();
+				$moduleContainer.stopThrobbing();
+			},
+			onErrorCallback: function () {
+				// TODO: handle failure
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)